### PR TITLE
Case sensitive column names in DI SQL destinations

### PR DESCRIPTION
**Pull request by Luishfs**
# Description

Users have reported not being able to use case sensitive column names on SQL destinations in Data Integration pipelines. ( Slack threads [example-1](https://mageai.slack.com/archives/C03HTTWFEKE/p1696280965183949) and [example-2](https://mageai.slack.com/archives/C03HTTWFEKE/p1695309486767869) )

To fix this, the optinal config `lower_case` was added. If set to `False`, the requested Destination will allow for case sensitive column names when executing all `clean_column_name` functions.

This PR adds this feature to the following destinations:

1. Snowflake
2. PostgreSQL
3. Redshift
4. OracleDB
5. MySQL
6. MSSQL
7. BigQuery

# How Has This Been Tested?

Unittests were created at mage_integrations/tests/destinations folder.
These tests cover build-table enforcing case sensitive queries.

# Checklist
- [X] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [X] I have performed a self-review of my own code
- [X] I have added unit tests that prove my fix is effective or that my feature works
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_integrations/mage_integrations/destinations/bigquery/__init__.py [hunk 1]
```diff
@@ -114,7 +119,7 @@ def build_create_table_commands(
             create_table_command = f'''
 {create_table_command}
 PARTITION BY
-  DATE(`{clean_column_name(partition_col)}`)
+  DATE(`{clean_column_name(partition_col, self.use_lowercase)}`)
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/destinations/bigquery/__init__.py:122]: I would also suggest define the clean_column_name method in the base sql destination to reduce the duplicate code.
```
def self. clean_column_name(col):
    return clean_column_name(col, lower_case=self. use_lowercase)
```

**Comment by Jacse** [comment]
As a side note: From our side we would want the lowercase-flag to apply both to column names and table names

File: mage_integrations/mage_integrations/destinations/bigquery/__init__.py [hunk 2]
```diff
@@ -64,6 +64,10 @@ class BigQuery(Destination):
     def quote(self) -> str:
         return '`'
 
+    @property
+    def use_lowercase(self) -> bool:
```
[the remark this answers is not in the document]
Comment (Luishfs) [reply mage_integrations/mage_integrations/destinations/bigquery/__init__.py]: Added in 6b629d234dfc02cd6157d07a31dd2e14d469f4c5

File: mage_integrations/mage_integrations/tests/destinations/bigquery/test_bigquery.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import unittest
+
+from mage_integrations.destinations.bigquery import BigQuery
+
+SCHEMA = {'properties': {
```
[the remark this answers is not in the document]
Comment (Luishfs) [reply mage_integrations/mage_integrations/tests/destinations/bigquery/test_bigquery.py]: Modified in 6b629d234dfc02cd6157d07a31dd2e14d469f4c5

File: mage_integrations/mage_integrations/tests/destinations/bigquery/test_bigquery.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+def bigquery_config():
+    return {
+        'lower_case': False
+    }
+
+
+class BigQueryDestinationTests(unittest.TestCase):
+
+    def test_create_table_commands(self):
+
```
[the remark this answers is not in the document]
Comment (Luishfs) [reply mage_integrations/mage_integrations/tests/destinations/bigquery/test_bigquery.py]: Removed in 6b629d234dfc02cd6157d07a31dd2e14d469f4c5

File: mage_integrations/mage_integrations/destinations/bigquery/__init__.py (hunk 1 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/destinations/bigquery/__init__.py]: Modified in 6b629d234dfc02cd6157d07a31dd2e14d469f4c5

MsSQL and OracleDB have their on `clean_column_name` methods in their respective utils.py, so they were not modified

**Pull request merged**